pkg/api/mlflow/dao/repositories/helpers.go

```diff
@@ -30,3 +33,22 @@ func makeParamConflictPlaceholdersAndValues(params []models.Param) (string, []in
 	}
 	return placeholders, valuesArray
 }
+
+// addJsonCondition adds condition(s) to the query to select items having the specified map of json paths
+// and values. Json path is expressed as "outerkey.nestedKey".
+func addJsonCondition(tx *gorm.DB, jsonColumnName string, jsonPathValueMap map[string]any) {
+	if len(jsonPathValueMap) == 0 {
+		return
+	}
+	switch tx.Dialector.Name() {
+	case postgres.Dialector{}.Name():
+		for k, v := range jsonPathValueMap {
+			path := strings.ReplaceAll(k, ".", ",")
+			tx.Where(fmt.Sprintf("%s#>>'{%s}' = ?", jsonColumnName, path), v)
+		}
+	default:
+		for k, v := range jsonPathValueMap {
+			tx.Where(fmt.Sprintf("%s->>'%s' = ?", jsonColumnName, k), v)
+		}
+	}
+}
```

Comment:
i think this will be awkward because `addJsonCondition` would have to return tuple and only the second is expanded, yes?

Reply:
I thought about this only because of the moment related to the tests.  but why awkward? functions can return `tuple` like that:
```
func asdasd() (query interface{}, args []interface{}){}
```
I don't know for me it looks a bit cleaner and testable.

Reply:
i think the current code is less awkward because the condition needs to be a string and a value for each context param -- so we'd have to make one big string and a values slice. I think for db-related functionality we should lean on the integration tests because that's the only way we know the sql is correct.

Reply:
no no, Im talking about `unit` tests for this part. that's why I want to get rid of `db` here.

Reply:
i think the code will be a lot less clean (collecting a large sql clause, collecting a values slice, returning both as a tuple) and then we still need to run it in a database (integration test) to have any confidence it's correct so the unit test is low value.